sktime/forecasting/base/convertIO/_convertIO.py

```diff
@@ -0,0 +1,190 @@
+
+-----------------------------------------
+
+Function signature of convert_to
+
+Parameters
+----------
+what : object to convert (any type)
+to_type : type, the type to convert "what" to
+as_scitype : str - name of scitype the object "what" is considered as
+store : reference of storage for lossy conversions, default=None (no store)
+
+Returns
+-------
+converted_what : to_type - object what converted to to_type
+
+Raises
+------
+TypeError if no suitable key in convert exists
+    i.e., (type(what), to_type, as_scitype)
+
+
+copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""
+
+
+__author__ = ["fkiraly"]
+
+
+import numpy as np
+import pandas as pd
+
+
+# dictionary indexed by triples of types
+#  1st element = convert from - type
+#  2nd element = convert to - type
+#  3rd element = considered as this scitype - string
+# elements are conversion functions of machine type (1st) -> 2nd
+
+convert = dict()
+
+
+def convert_identity(what, store=None):
+
+    return what
+
+
+# assign identity function to type conversion to self
+for tp in [pd.Series, pd.DataFrame, np.array]:
+    convert[(tp, tp, "Series")] = convert_identity
+
+
+def convert_UvS_to_MvS_as_Series(what: pd.Series, store=None) -> pd.DataFrame:
+
+    if not isinstance(what, pd.Series):
+        raise TypeError("input must be a pd.Series")
+
+    if isinstance(store, dict) and "cols" in store.keys and len(store["cols"]) == 1:
+        res = pd.DataFrame(what, columns=store["cols"])
+    else:
+        res = pd.DataFrame(what)
+
+    return res
+
+
+convert[(pd.Series, pd.DataFrame, "Series")] = convert_UvS_to_MvS_as_Series
+
+
+def convert_MvS_to_UvS_as_Series(what: pd.DataFrame, store=None) -> pd.Series:
+
+    if not isinstance(what, pd.DataFrame):
+        raise TypeError("input is not a pd.DataFrame")
+
+    if len(what.columns) != 1:
+        raise ValueError("pd.DataFrame must be pd.DataFrame with one column")
+
+    if isinstance(store, dict):
+        store["cols"] = what.columns[[0]]
+
+    return what[what.columns[0]]
+
+
+convert[(pd.Series, pd.DataFrame, "Series")] = convert_MvS_to_UvS_as_Series
+
+
+def convert_MvS_to_np_as_Series(what: pd.DataFrame, store=None) -> np.array:
+
+    if not isinstance(what, pd.DataFrame):
+        raise TypeError("input must be a pd.DataFrame")
+
+    if isinstance(store, dict):
+        store["cols"] = what.columns
+
+    return what.to_numpy()
+
+
+convert[(pd.DataFrame, np.array, "Series")] = convert_MvS_to_np_as_Series
+
+
+def convert_UvS_to_np_as_Series(what: pd.Series, store=None) -> np.array:
+
+    if not isinstance(what, pd.Series):
+        raise TypeError("input must be a pd.Series")
+
+    return pd.DataFrame(what).to_numpy()
+
+
+convert[(pd.Series, np.array, "Series")] = convert_UvS_to_np_as_Series
+
+
+def convert_np_to_MvS_as_Series(what: np.array, store=None) -> pd.DataFrame:
+
+    if not isinstance(what, np.array) and len(what.shape) != 2:
```

Comment:
just checking that you have explicitly chosen to exclude 1D numpy arrays?

It is feasible a user tries to use a multivariate forecaster to do univariate forecasting with a 1D numpy input, and the current form of code would not allow that. I have no strong preference either way, but thought it is checking if this is what you intended.

Reply:
>just checking that you have explicitly chosen to exclude 1D numpy arrays?

No, but I did not want to complicate an already difficult discussion.
With this design, they would be easy to add, but let's not mix the "should we support 1D numpy arrays" discussion into this. See also #897 for the corresponding conversion.

Reply:
nothing to resolve since clarification